src/reducers/versions.tsx

```diff
@@ -1523,17 +1542,33 @@ export const createReducer = ({
       }
       case getType(actions.setCurrentVersionId): {
         const { versionId } = action.payload;
-        let { expandedPaths } = state;
+        let { expandedPaths, versionInfo } = state;
+        const { selectedPath, visibleSelectedPath } = state;
         const version = getVersionInfo(state, versionId);
 
         if (version && expandedPaths) {
           expandedPaths = _getPrunedExpandedPaths(expandedPaths, version);
         }
 
+        // TODO: remove this conditional statement once no code depends
+        // on `selectedPath` and `visibleSelectedPath` in version
+        // See https://github.com/mozilla/addons-code-manager/issues/1258
+        if (version) {
+          versionInfo = {
+            ...versionInfo,
+            [version.id]: {
+              ...version,
+              selectedPath: selectedPath || version.selectedPath,
```

Comment:
> I realized that the `selectedPath` may not exist in the new version, hoping this bug will not block this patch (I am also not sure what we want in this case).

Ah, yes, good catch. I suggest modifying `getPrunedExpandedPaths()` to be `removeNonExistantVersionPaths()` or something so you can call it here like `removeNonExistantVersionPaths({ paths: [selectedPath], version })` (with a fall back to `version.selectedPath`). Maybe there is a better way to create the shared function but that's the gist of it.

It shouldn't be such a big change (I hope) but you could maybe try it first and if it's a big change then add a todo comment and file an issue to address it in a future patch.

Reply:
How do you think about the following approach?
```diff
diff --git a/src/reducers/versions.tsx b/src/reducers/versions.tsx
index 61b77d2..85fe2a0 100644
--- a/src/reducers/versions.tsx
+++ b/src/reducers/versions.tsx
@@ -485,12 +485,16 @@ export const getVersionInfo = (versions: VersionsState, versionId: number) => {
   return versions.versionInfo[versionId];
 };
 
+export const doesPathExist = (path: string, version: Version) => {
+  const paths = new Set(version.entries.map((e) => e.path));
+  return paths.has(path);
+};
+
 export const getPrunedExpandedPaths = (
   expandedPaths: string[],
   version: Version,
 ) => {
-  const paths = new Set(version.entries.map((e) => e.path));
-  return expandedPaths.filter((p) => paths.has(p));
+  return expandedPaths.filter((p) => doesPathExist(p, version));
 };
 
 export const getEntryStatusMap = ({
@@ -1558,7 +1562,10 @@ export const createReducer = ({
             ...versionInfo,
             [version.id]: {
               ...version,
-              selectedPath: selectedPath || version.selectedPath,
+              selectedPath:
+                selectedPath && doesPathExist(selectedPath, version)
+                  ? selectedPath
+                  : version.selectedPath,
               visibleSelectedPath,
             },
           };
```

Reply:
I just tried to avoid using `removeNonExistantVersionPaths({ paths: [selectedPath], version }).length = 0`  to determine if `selectedPath` exists in `version`, since it looks not that nice.

Reply:
Yeah, that looks better but I suggest introducing two functions so you can re-use it in an efficient way, like:

```tsx
export const getPrunedExpandedPaths = (
  expandedPaths: string[],
  version: Version,
) => {
  const doesPathExist = createVersionPathChecker(version);
  return expandedPaths.filter(doesPathExist);
};

export const versionPathExists = (path: string, version: Version) => {
  const doesPathExist = createVersionPathChecker(version);
  return doesPathExist(path);
}
```

Version file lists typically get quite large because of all the nested locale files.